---
title: "Differential expression analysis"
author: "Andreas Norlin"
date: "2025-01-14"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

rm(list = ls())

package.list <- c("dplyr","tidyverse","ggplot2",
                  "lemon","vegan","Rmisc",
                  "lubridate","patchwork","stringi",
                  "RColorBrewer","pheatmap")

if(!all(package.list %in% installed.packages()[,"Package"])){
  install.packages(package.list[!(package.list %in% installed.packages()[,"Package"])])
}

bioc.packages <- c("phyloseq","DESeq2")

if(!all(bioc.packages %in% installed.packages()[,"Package"])){
  if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install(bioc.packages[!(bioc.packages %in% installed.packages()[,"Package"])])
}

library(dplyr)
library(tidyverse)
library(ggplot2)
library(lemon)
library(vegan)
library(Rmisc)
library(lubridate)
library(RColorBrewer)
library(pheatmap)

library(phyloseq)
library(patchwork)
library(DESeq2)
library(stringi)

TPM <- FALSE

if(TPM == TRUE){
  load("CSV_files/Subsections.RData",verbose = TRUE)
}else{
  load("CSV_files/Subsections_numreads.RData",verbose = TRUE)
}

```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r, Data setup}

# Input_data <- Bact.Sub
Input_data <- rbind(Euk.Sub,Prok.Sub)

for(i in 1:nrow(Input_data)){
  if(i == 1){
    n_KEGG <- length(unlist(strsplit(Input_data$KEGG_ko[i],",",fixed = TRUE)))
  }
  m_KEGG <- length(unlist(strsplit(Input_data$KEGG_ko[i],",",fixed = TRUE)))
  
  if(m_KEGG > n_KEGG){
    n_KEGG <- m_KEGG
  }
}

KEGG_columns <- paste("KEGG",1:n_KEGG,sep = "")

Input_data <- separate(Input_data,col = KEGG_ko,into = KEGG_columns,
                     sep = ",", remove = FALSE, fill = "right", extra = "drop")

KO_paths_Bact <- read.csv("KO_pathway.csv")
KO_paths_Euk <- read.csv("KO_pathway_Euk.csv")

KO_paths <- rbind(KO_paths_Euk,
                  KO_paths_Bact[!(KO_paths_Bact$K1 %in% KO_paths_Euk$K1),])

Input_data$K1_paths <- unlist(lapply(Input_data$KEGG1,
                                     function(x){
                                       ifelse(x == "-","",KO_paths$ko[KO_paths$K1 == x])}))

# Input_data <- Input_data[!is.na(Input_data$Class),]
Input_data$Domain_K1 <- paste(Input_data$Domain,Input_data$KEGG1,sep = "_")

source("summary_count.R")

day.vec <- paste(meta.data$SampleID,"_quant",sep = "")

df.count <- summary_count(Input_data,Col_vec = day.vec,Col_filter = "Domain_K1")

names(df.count) <- unlist(lapply(names(df.count),
                                 function(x){unlist(strsplit(x,
                                                             split = "_",
                                                             fixed = TRUE))[1]}))

df_meta <- meta.data[,c("SampleID","SampleName","AssemblyGroup","Day.num","ToD")] %>%
  mutate(ToD = factor(ToD,levels = c("morning","afternoon","evening","night")))


# Optional extra step to average the replicates of each day ####
AvgDays <- F

if(AvgDays == TRUE){
  # Input_data <- Bact.Sub
  
  Input_data <- df.count
  
  vec_assembly <- unique(meta.data$AssemblyGroup)
  
  for(i in 1:length(vec_assembly)){
    if(i == 1){
  
      df_means <- row.names(Input_data)
    }
  
    df_temp <- Input_data[,meta.data$SampleID[meta.data$AssemblyGroup == vec_assembly[i]]]
  
    meta_temp <- df_meta %>%
      distinct(AssemblyGroup,.keep_all = TRUE) %>%
      filter(AssemblyGroup == vec_assembly[i])
  
    vec <- rowMeans(df_temp)
    # vec <- rowSums(df_temp)
    df_means <- cbind.data.frame(df_means,vec)
  
    names(df_means)[i+1] <- vec_assembly[i]
  }
  
  # creating factors for the ToD in order to analyse the time points chronologically
  df_meta <- df_meta %>%
    distinct(AssemblyGroup,.keep_all = TRUE)
  
  df.count <- df_means[,-1]
  rm(list = c("df_means"))
}

# Rounding data to become integer ####

df.0filtered <- df.count[apply(df.count, 1, function(x){sum(x == 0)}) < 24,]

if(TPM == TRUE){
 countData <- round(df.0filtered*1000) 
}else{
  countData <- round(df.0filtered)
}

countData <- cbind.data.frame("SequenceID" = row.names(df.0filtered),countData)

head(countData)
tail(countData)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
Day_toggle <- "ToD"

GroupOI <- c("morning","afternoon","evening","night")#[c(1,3)]

if(AvgDays == TRUE){
  filter <- c("SequenceID",df_meta$AssemblyGroup[df_meta$ToD %in% GroupOI])
} else {
  filter <- c("SequenceID",df_meta$SampleID[df_meta$ToD %in% GroupOI]) 
}

TaxFilter <- Input_data$Domain == "Archaea"

if(Day_toggle == "ToD"){
  dds <- DESeqDataSetFromMatrix(countData=countData[TaxFilter,filter],
                              colData=df_meta[df_meta$ToD %in% GroupOI,], 
                              design=~ToD, tidy = TRUE)
} else if(Day_toggle == "Day.num"){
  dds <- DESeqDataSetFromMatrix(countData=countData[TaxFilter,filter],
                              colData=df_meta[df_meta$ToD %in% GroupOI,], 
                              design=~Day.num, tidy = TRUE)
}

dds

# PCA analysis #####-----------------------------------------------------------

vsdata <- varianceStabilizingTransformation(dds, blind = FALSE)

pcaData <- plotPCA(vsdata,
                   intgroup=c("Day.num","ToD"),
                   returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Day.num, shape=ToD)) +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  geom_point(size=3) +
  labs(x = paste0("PC1: ",percentVar[1],"% variance"),
       y = paste0("PC2: ",percentVar[2],"% variance"))+
  # xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  # ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed() +
  scale_colour_brewer(type="qual", palette="Dark2")+
  theme_classic()

# ggsave("DE_plots/AnnotatedContigs_PCA.png",dpi = 300,width = 8,height = 6)
```

```{r pressure, echo=FALSE}
# Running vst and plotting pair wise PCAs
load("CSV_files/salmon.merged.numreads.subset5.RData",verbose = T)

df_meta <- meta.data[,c("SampleID","SampleName","AssemblyGroup","Day.num","ToD")] %>%
  mutate(ToD = factor(ToD,levels = c("morning","afternoon","evening","night")))

countData <- cbind.data.frame("SequenceID" = salmon.numreads[,1],round(salmon.numreads[,-c(1,26)]))
# countData <- countData[1:1000000,]

n <- 1

for(ToD1 in 1:4){
  for(ToD2 in 1:4){
    if(ToD1 < ToD2){
      
      Day_toggle <- "ToD"

      GroupOI <- c("morning","afternoon","evening","night")[c(ToD1,ToD2)]
      
      # if(AvgDays == TRUE){
      #   filter <- c("SequenceID",df_meta$AssemblyGroup[df_meta$ToD %in% GroupOI])
      # } else {
      #   filter <- c("SequenceID",df_meta$SampleID[df_meta$ToD %in% GroupOI]) 
      # }
      
      filter <- c("SequenceID",df_meta$SampleID[df_meta$ToD %in% GroupOI]) 
      
      if(Day_toggle == "ToD"){
        dds <- DESeqDataSetFromMatrix(countData=countData[,filter],
                                    colData=df_meta[df_meta$ToD %in% GroupOI,], 
                                    design=~ToD, tidy = TRUE)
      } else if(Day_toggle == "Day.num"){
        dds <- DESeqDataSetFromMatrix(countData=countData[,filter],
                                    colData=df_meta[df_meta$ToD %in% GroupOI,], 
                                    design=~Day.num, tidy = TRUE)
      }
      
      dds
      
      # PCA analysis #####-----------------------------------------------------------
      
      vsdata <- varianceStabilizingTransformation(dds, blind = FALSE)
      
      pcaData <- plotPCA(vsdata,
                         intgroup=c("Day.num","ToD"),
                         returnData = TRUE)
      percentVar <- round(100 * attr(pcaData, "percentVar"))
      PCA_plot <- ggplot(pcaData, aes(PC1, PC2, color=Day.num, shape=ToD)) +
        geom_vline(xintercept = 0) +
        geom_hline(yintercept = 0) +
        geom_point(size=3) +
        labs(x = paste0("PC1: ",percentVar[1],"% variance"),
             y = paste0("PC2: ",percentVar[2],"% variance"),
             title = paste0(GroupOI[1]," vs ",GroupOI[2]))+
        coord_fixed() +
        scale_colour_brewer(type="qual", palette="Dark2")+
        theme_classic()
      
      print(PCA_plot)
      
      pcaData$ToD1 <- GroupOI[1]
      pcaData$ToD2 <- GroupOI[2]
      
      if(n == 1){
        dfOut <- pcaData
      }else{
        dfOut <- rbind.data.frame(dfOut,pcaData) 
      }
      n <- n+1
    }
  }
}

ggplot(dfOut, aes(PC1, PC2, color=Day.num, shape=ToD)) +
  geom_vline(xintercept = 0) +
  geom_hline(yintercept = 0) +
  geom_point(size=3) +
  labs(x = paste0("PC1"),
       y = paste0("PC2"),
       title = "PCA plots")+
  facet_rep_grid(factor(ToD1, levels = c("morning","afternoon","evening")) ~ factor(ToD2,levels = c("afternoon","evening","night")))+
  scale_colour_brewer(type="qual", palette="Dark2")+
  theme_classic()

ggsave("DE_plots/PairWise_PCA.png",dpi = 300,height = 6,width = 8)
```

```{r pressure, echo=FALSE}

load("CSV_files/salmon.merged.numreads.subset5.RData",verbose = T)

df_meta <- meta.data[,c("SampleID","SampleName","AssemblyGroup","Day.num","ToD")] %>%
  mutate(ToD = factor(ToD,levels = c("morning","afternoon","evening","night")))

countData <- cbind.data.frame("SequenceID" = salmon.numreads[,1],round(salmon.numreads[,-c(1,26)]))

# rdv <- sample(nrow(salmon.numreads),10000,replace = F)
# countData <- countData[rdv,]

pAdj_thresh <- 0.05

for(ToD1 in 1:4){
  if(ToD1 == 1){
    time1 <- Sys.time()
    
    n <- 1
  }
  
  for(ToD2 in 1:4){
    if(ToD1 < ToD2){
      
      Day_toggle <- "ToD"

      GroupOI <- c("morning","afternoon","evening","night")[c(ToD1,ToD2)]
      
      print(paste("Run",n,"comparing",GroupOI[1],"against",GroupOI[2]))
      
      # if(AvgDays == TRUE){
      #   filter <- c("SequenceID",df_meta$AssemblyGroup[df_meta$ToD %in% GroupOI])
      # } else {
      #   filter <- c("SequenceID",df_meta$SampleID[df_meta$ToD %in% GroupOI]) 
      # }
      
      filter <- c("SequenceID",df_meta$SampleID[df_meta$ToD %in% GroupOI])
      
      print("Creating dds object")
      
      if(Day_toggle == "ToD"){
        dds <- DESeqDataSetFromMatrix(countData=countData[,filter],
                                    colData=df_meta[df_meta$ToD %in% GroupOI,], 
                                    design=~ToD, tidy = TRUE)
      } else if(Day_toggle == "Day.num"){
        dds <- DESeqDataSetFromMatrix(countData=countData[,filter],
                                    colData=df_meta[df_meta$ToD %in% GroupOI,], 
                                    design=~Day.num, tidy = TRUE)
      }
      
      # PCA analysis #####-----------------------------------------------------------
      print("Running DESeq on the dds object")
      
      dds <- DESeq(dds)
  
      # Take a look at the results table --------------------------------------------
      print("Running results on the dds after DESeq()")
      
      res <- results(dds)
      
      # Sort summary list by p-value ------------------------------------------------
      
      res <- res[order(res$padj,decreasing = F),]
      
      x_range <- min(max(abs(res$log2FoldChange),na.rm = TRUE),150)
      y_range <- min(max(-log10(res$pvalue),na.rm = TRUE),150)
      
      par(mfrow=c(2,3))
    
      for(i in 1:6){
        plotCounts(dds,
                   gene=row.names(res)[i],
                   intgroup="ToD")
      }
    
      mtext(paste(GroupOI[1],"vs",GroupOI[2]), side = 3, line = -1.5, outer = TRUE)
      
      #reset par
      par(mfrow=c(1,1))
      # Make a basic volcano plot
      with(res,
           plot(log2FoldChange,
                -log10(pvalue),
                pch=20,
                main=paste0(GroupOI[1]," vs ",GroupOI[2]," Volcano plot"),
                ylim=c(0,y_range),
                xlim=c(-x_range,x_range)))
      
      # Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
      with(subset(res,
                  padj<pAdj_thresh ),
           points(log2FoldChange,-log10(pvalue),
                  pch=20,
                  col="blue"))
      
      with(subset(res,
                  padj<pAdj_thresh & abs(log2FoldChange)>1 & !is.na(padj<pAdj_thresh)),
           points(log2FoldChange, -log10(pvalue),
                  pch=20,
                  col="red"))
      
      res$colorGroup <- "insig"
      res$colorGroup[res$padj < pAdj_thresh &
                             !is.na(res$padj<pAdj_thresh)] <- "sig"
      
      res$colorGroup[res$padj < pAdj_thresh &
                             !is.na(res$padj<pAdj_thresh) &
                             abs(res$log2FoldChange) >1] <- "sigLarge"
      
      res$ToD1 <- GroupOI[1]
      res$ToD2 <- GroupOI[2]
      
      if(n == 1){
       dfOut <- res
      }else{
       dfOut <- rbind(dfOut,res)
      }
      
      print(paste("Run",n,"finished.",round(difftime(Sys.time(),time1,units = "hour"),2),"hours have elapsed"))
      n <- n+1
    }
  }
}

DE_plot <- ggplot(dfOut[!is.na(dfOut$padj),])+
  geom_vline(xintercept = c(-1,1))+
  geom_point(aes(x = log2FoldChange,
                 y = -log(pvalue,10),
                 color = colorGroup))+
  facet_rep_grid(factor(ToD1, levels = c("morning","afternoon","evening")) ~ factor(ToD2,levels = c("afternoon","evening","night")))+
  scale_color_manual(values = c("insig" = "black",
                                "sig" = "blue",
                                "sigLarge" = "red"))+
  theme_classic(base_size = 12)+
  theme(legend.title=element_blank(),
        legend.position = "right",
        legend.key.size = unit(1.5, "cm"),
        axis.text = element_text(colour = "black"),
        axis.ticks = element_line(colour = "black"),
        panel.border = element_rect(linewidth = 1,
                                    colour = "black",
                                    fill = NA),
        strip.background.x =  element_rect(colour = NA,
                                           fill = NA),
        strip.text.x = element_text(face = "bold",
                                    size = 20),
        strip.text.y = element_text(face = "bold",
                                    size = 15))

ggsave(filename = "DE_plots/PairWise_DE.png",plot = DE_plot,dpi = 300,height = 9,width = 12)
```

```{r pressure, echo=FALSE}
# Now we’re ready to run DESEQ function ---------------------------------------

dds <- DESeq(dds)

# Take a look at the results table --------------------------------------------

res <- results(dds)

head(results(dds, tidy = TRUE))

# Summary of differential gene expression -------------------------------------

summary(res) #summary of results

# Sort summary list by p-value ------------------------------------------------

res <- res[order(res$padj,decreasing = F),]
head(res)

par(mfrow=c(2,3))

for(i in 1:6){
  plotCounts(dds,
             gene=row.names(res)[i],
             intgroup=Day_toggle)
}

pAdj_thresh <- 0.01

x_range <- min(max(abs(res$log2FoldChange),na.rm = TRUE),150)
y_range <- min(max(-log10(res$pvalue),na.rm = TRUE),150)

#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res,
     plot(log2FoldChange,
          -log10(pvalue),
          pch=20,
          main="Volcano plot",
          ylim=c(0,y_range),
          xlim=c(-x_range,x_range)))


# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res,
            padj<pAdj_thresh ),
     points(log2FoldChange,-log10(pvalue),
            pch=20,
            col="blue"))

with(subset(res,
            padj<pAdj_thresh & abs(log2FoldChange)>1 & !is.na(padj<pAdj_thresh)),
     points(log2FoldChange,-log10(pvalue),
            pch=20,
            col="red"))

dfOut <- res[res$padj<pAdj_thresh & !is.na(res$padj<pAdj_thresh),]
```


```{r}
dfOut$ko <- row.names(dfOut)

source("pathway_API.R")

vec <- rep(NA,nrow(dfOut))

for(i in 1:nrow(dfOut)){
  vec[i] <- pathway_API(dfOut$ko[i])$name
  
  if(is.na(vec[i])){
    vec[i] <- dfOut$ko[i]
  }
}

row.names(dfOut) <- vec
# dfOut <- dfOut[order(abs(dfOut$log2FoldChange),decreasing = TRUE),]

n <- min(nrow(dfOut),6)
par(mfrow=c(2,3))

for(i in 1:n){
  plotCounts(dds,
             gene=dfOut$ko[i],
             intgroup=Day_toggle,
             main = row.names(dfOut)[i])
}
```

```{r}
source("pathway_API.R")

if(AvgDays == TRUE){
  filter <- c("SequenceID",df_meta$AssemblyGroup[df_meta$ToD %in% GroupOI])
} else {
  filter <- c("SequenceID",df_meta$SampleID[df_meta$ToD %in% GroupOI]) 
}

GroupOI <- c("morning","afternoon","evening","night")

pAdj_thresh <- 0.01

for(j in 1:4){
  testGroup_vec <- GroupOI[j]
  
  df_meta$testGroup <- unlist(lapply(df_meta$ToD,function(x){ifelse(x == testGroup_vec,testGroup_vec,"other")}))
  
  df_meta <- df_meta %>%
    mutate(testGroup = factor(testGroup,levels = c("other",testGroup_vec)))
  
  dds <- DESeqDataSetFromMatrix(countData=countData[,filter],
                                colData=df_meta[df_meta$ToD %in% GroupOI,], 
                                design=~testGroup, tidy = TRUE)
  
  # Now we’re ready to run DESEQ function ---------------------------------------
  
  dds <- DESeq(dds)
  
  # Take a look at the results table --------------------------------------------
  
  res <- results(dds)
  
  head(results(dds, tidy = TRUE))
  
  # Summary of differential gene expression -------------------------------------
  
  summary(res) #summary of results
  
  # Sort summary list by p-value ------------------------------------------------
  
  res <- res[order(res$padj,decreasing = F),]
  head(res)
  
  x_range <- min(max(abs(res$log2FoldChange),na.rm = TRUE),150)
  y_range <- min(max(-log10(res$pvalue),na.rm = TRUE),10)
  
  par(mfrow=c(2,3))

  for(i in 1:6){
    plotCounts(dds,
               gene=row.names(res)[i],
               intgroup="testGroup")
  }

  mtext(testGroup_vec, side = 3, line = -1.5, outer = TRUE)
  
  #reset par
  par(mfrow=c(1,1))
  # Make a basic volcano plot
  with(res,
       plot(log2FoldChange,
            -log10(pvalue),
            pch=20,
            main=paste0(testGroup_vec," Volcano plot"),
            ylim=c(0,y_range),
            xlim=c(-x_range,x_range)))
  
  # Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
  with(subset(res,
              padj<pAdj_thresh ),
       points(log2FoldChange,-log10(pvalue),
              pch=20,
              col="blue"))
  
  with(subset(res,
              padj<pAdj_thresh & abs(log2FoldChange)>1 & !is.na(padj<pAdj_thresh)),
       points(log2FoldChange, -log10(pvalue),
              pch=20,
              col="red"))
  
  
  dfOut.temp <- res[res$padj<pAdj_thresh & !is.na(res$padj<pAdj_thresh),]
  
  if(dim(dfOut.temp)[1] != 0){
   dfOut.temp$testGroup <- testGroup_vec
   
   # dfOut.temp$ko <- row.names(dfOut.temp)
   dfOut.temp$Domain <- unlist(lapply(row.names(dfOut.temp),
                                  function(x){unlist(strsplit(x,
                                                              split = "_",
                                                              fixed = TRUE))[1]}))
   dfOut.temp$ko <- unlist(lapply(row.names(dfOut.temp),
                                  function(x){unlist(strsplit(x,
                                                              split = "_",
                                                              fixed = TRUE))[2]}))

   vec <- rep(NA,nrow(dfOut.temp))

   for(i in 1:nrow(dfOut.temp)){
     vec[i] <- pathway_API(dfOut.temp$ko[i])$name

     if(is.na(vec[i])){
       vec[i] <- dfOut.temp$ko[i]
     }
   }

   row.names(dfOut.temp) <- vec
  }
  
  res$testGroup <- testGroup_vec
  
  if(j != 1){
    dfOut <- rbind(dfOut,dfOut.temp)
    dfOut_all <- rbind(dfOut_all,res)
  }else{
    dfOut <- dfOut.temp
    dfOut_all <- res
  }
}


```


```{r}

dfOut_all <- dfOut_all[!is.na(dfOut_all$pvalue),]

dfOut_all$Domain <- unlist(lapply(row.names(dfOut_all),
                                  function(x){unlist(strsplit(x,
                                                              split = "_",
                                                              fixed = TRUE))[1]}))

dfOut_all$colorGroup <- "insig"
dfOut_all$colorGroup[dfOut_all$padj < pAdj_thresh &
                       !is.na(dfOut_all$padj<pAdj_thresh)] <- "sig"

# dfOut_all$colorGroup[dfOut_all$padj < pAdj_thresh &
#                        !is.na(dfOut_all$padj<pAdj_thresh) &
#                        abs(dfOut_all$log2FoldChange) >1] <- "sigLarge"

ggplot(dfOut_all)+
  geom_vline(xintercept = c(-1,1))+
  geom_point(aes(x = log2FoldChange,
                 y = -log(pvalue,10),
                 color = Domain,
                 shape = colorGroup))+
  facet_wrap(facets = "testGroup",
             nrow = 2, ncol = 2,
             scale = "free_y")+
  scale_shape_manual(values = c("insig" = 46,
                       "sig" = 16,
                       "sigLarge" = 16))+
  scale_color_manual(values = c("Bacteria" = "dodgerblue",
                                "Eukaryota" = "goldenrod"))+
  theme_classic(base_size = 16)+
  theme(legend.title=element_blank(),
        legend.position = "right",
        legend.key.size = unit(1.5, "cm"),
        axis.text = element_text(colour = "black"),
        axis.ticks = element_line(colour = "black"),
        panel.border = element_rect(linewidth = 1,
                                    colour = "black",
                                    fill = NA),
        strip.background.x =  element_rect(colour = NA,
                                           fill = NA),
        strip.text.x = element_text(face = "bold",
                                    size = 20),
        strip.text.y = element_text(face = "bold",
                                    size = 15))
```


```{r}
df_UpDown <- matrix(NA,nrow = length(unique(dfOut$ko)),ncol = 4)
colnames(df_UpDown) <- c("morning","afternoon","evening","night")
rownames(df_UpDown) <- unique(dfOut$ko)

for(i in 1:4){
  df.temp <- dfOut[dfOut$testGroup == colnames(df_UpDown)[i],]
  
  for(j in 1:nrow(df_UpDown)){
   vec <- ifelse(rownames(df_UpDown)[j] %in% df.temp$ko,
                 ifelse(df.temp$log2FoldChange[df.temp$ko == rownames(df_UpDown)[j]]<0,-1,1),
                 0)
    
    # vec <- ifelse(rownames(df_UpDown)[j] %in% df.temp$ko,
    #              df.temp$log2FoldChange[df.temp$ko == rownames(df_UpDown)[j]],
    #              0)
   
   df_UpDown[j,i] <- vec
  }
}

library(pheatmap)

Class_df <- data.frame(KO_paths[KO_paths$K1 %in% dfOut$ko,2])
names(Class_df) <- c("Pathway")
rownames(Class_df) <- KO_paths[KO_paths$K1 %in% dfOut$ko,1]

Class_df[Class_df[,1] == "",1] <- "-"

MetaData_df <- data.frame(matrix(GroupOI,ncol = 1))
names(MetaData_df) <- "ToD"
rownames(MetaData_df) <- GroupOI

for(i in 1:nrow(Class_df)){
  vec <- pathway_API(Class_df[i,1])$name
  
  if(is.na(vec)){
    vec <- "-"
  }
  
  Class_df[i,1] <- vec
}

Pathway_cols <- c("white","#D95F02","#7570B3","#E7298A",
                  "#66A61E","#E6AB02","#A6761D","#7FC97F",
                  "#BEAED4","#FDC086","#FFFF99","#386CB0",
                  "#F0027F","#BF5B17","darkblue","black",
                  "#E41A1C","#377EB8","#4DAF4A","#984EA3")
names(Pathway_cols) <- unique(Class_df[,1])
# names(Pathway_cols) <- top10_all

Count_cols <- list("Pathway" = Pathway_cols,
                   # "Class" = Class_cols,
                   # "Day" = c("Day1" = "darkgreen",
                   #            "Day2" = "blueviolet"),
                   "ToD" = c("afternoon" = "#F7F4B7",
                                     "night" = "darkblue",
                                     "morning" = "#B7E1DD",
                                     "evening" = "orange"))

heat_plot <- pheatmap((df_UpDown),
             col = brewer.pal(3, 'RdBu'),
             cluster_rows = T, cluster_cols = T, # set to FALSE if you want to remove the dendograms
             clustering_distance_cols = 'euclidean',
             clustering_distance_rows = 'euclidean',
             clustering_method = 'ward.D',
             annotation_row = Class_df,
             annotation_names_row = T,
             annotation_col = MetaData_df,
             annotation_names_col = T,
             annotation_colors = Count_cols,
             show_rownames = F, show_colnames = F,
                          fontsize_row = 7,
                          fontsize_col = 5,
             )

# name_vec <- "Heatmaps/Bact_UpDown_pathways.pdf"
# 
# pdf(name_vec, height = 25, width = 20)
# heat_plot
# dev.off()

```


